src/service_control_client_impl.cc

```diff
@@ -140,16 +154,28 @@ void ServiceControlClientImpl::AllocateQuotaFlushCallback(
 
   quota_transport_(*quota_request_copy, quota_response,
                    [this, quota_request_copy, quota_response](Status status) {
-                     if (!status.ok()) {
+                     if (status.ok()) {
+                       (void)this->quota_aggregator_->CacheResponse(
+                         *quota_request_copy, *quota_response);
+                     } else if (IsQuotaFailOpenError(status)) {
+                       // Cache dummy response for fail open.
                        GOOGLE_LOG(ERROR) << "Failed in AllocateQuota call: "
-                                         << status.message();
-                       // cache dummy response for fail open
+                                         << status.message()
+                                         << ", but treated as fail-open";
                        AllocateQuotaResponse dummy_response;
                        (void)this->quota_aggregator_->CacheResponse(
-                           *quota_request_copy, dummy_response);
+                         *quota_request_copy, dummy_response);
                      } else {
+                       // Cache error response for fail close.
+                       GOOGLE_LOG(ERROR) << "Failed in AllocateQuota call: "
```

Comment:
We can remove this log line or change to VLOG(3)

Any errors that are due to invalid user input in the data plane are generally only DEBUG logs. We don't want malicious user to be able to generate lots of logs with bad input.

Anyways, I think we will log this in service control filter